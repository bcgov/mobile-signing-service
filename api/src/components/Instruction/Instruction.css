.instructions h3 {
  font-size: 20px;
  color: #00316b;
  margin: 1.5em 0 1em;
  border-bottom: 1px solid #ddd;
  padding-bottom: 0.3em;
}

.instructions ol {
  counter-reset: li;
  list-style: none;
  position: relative;
  padding-bottom: 10px;
}

.instructions ol li {
  padding: 5px 0 5px 60px;
  position: relative;
  margin-bottom: 20px;
}

.instructions ol li::before {
  content: counter(li);
  counter-increment: li;
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 30px;
  padding: 0 10px 0 0;
  color: #999;
  font-size: 22px;
  font-weight: 700;
  line-height: 30px;
  text-align: right;
  border-right: 1px solid #ddd;
}

.instructions p {
  margin: 5px 1em 15px;
}

.instructions pre {
  color: white;
  background-color: #494848;
}
