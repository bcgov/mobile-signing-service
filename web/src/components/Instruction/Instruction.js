import React from 'react';
import { XML_SAMPLES } from '../../constants';
import Toggle from '../Toggle/Toggle';
import './Instruction.css';

const Instruction = () => {
  return (
    <div className="instructions">
      <h3>Steps to Upload Your App</h3>
      <Toggle title="XcodeArchive">
        <p>To package up an xcarchive to submit for signing you need to:</p>
        <ol>
          <li>
            <p>Create a folder to hold the xcarchive and options.plist</p>
          </li>
          <li>
            <p>Copy the xcarchive from xcode into the folder from step 1</p>
          </li>
          <li>
            <p>
              Create or copy your options.plist from step 1, you could update the content from
              sample below:
            </p>
            <pre>{XML_SAMPLES.PLIST}</pre>
          </li>
          <li>
            <p>ZIP up the folder for submission</p>
          </li>
        </ol>
      </Toggle>
      <Toggle title="IPA">
        <p>Signing IPA files is currently Alpha. Please use xcarchive.</p>
      </Toggle>
      <Toggle title="APK">
        <p>To package up an android package to submit for signing you need to:</p>
        <ol>
          <li>
            <p>Select RELEASE mode in the Build Variant, and build APK</p>
          </li>
          <li>
            <p>Drag and drop the apk</p>
          </li>
        </ol>
      </Toggle>
    </div>
  );
};

export default Instruction;
