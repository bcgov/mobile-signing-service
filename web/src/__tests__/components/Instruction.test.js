import React from 'react';
import { shallow } from 'enzyme';
import Instruction from '../../components/Instruction/Instruction';

describe('Instruction Component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Instruction />);
    expect(wrapper).toMatchSnapshot();
  });
});
