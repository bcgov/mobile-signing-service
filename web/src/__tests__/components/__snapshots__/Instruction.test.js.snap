// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instruction Component matches snapshot 1`] = `
<div
  className="instructions"
>
  <h3>
    Steps to Upload Your App
  </h3>
  <Toggle
    title="XcodeArchive"
  >
    <p>
      To package up an xcarchive to submit for signing you need to:
    </p>
    <ol>
      <li>
        <p>
          Create a folder to hold the xcarchive and options.plist
        </p>
      </li>
      <li>
        <p>
          Copy the xcarchive from xcode into the folder from step 1
        </p>
      </li>
      <li>
        <p>
          Create or copy your options.plist from step 1, you could update the content from sample below:
        </p>
        <pre>
          
  &lt;?xml version="1.0" encoding="UTF-8"?&gt;
  &lt;!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd"&gt;
  &lt;plist version="1.0"&gt;
    &lt;dict&gt;
      &lt;key&gt;method&lt;/key&gt;
      &lt;string&gt;app-store&lt;/string&gt;
      &lt;key&gt;signingStyle&lt;/key&gt;
      &lt;string&gt;automatic&lt;/string&gt;
      &lt;key&gt;stripSwiftSymbols&lt;/key&gt;
      &lt;true/&gt;
      &lt;key&gt;uploadSymbols&lt;/key&gt;
      &lt;true/&gt;
    &lt;/dict&gt;
  &lt;/plist&gt;
  
        </pre>
      </li>
      <li>
        <p>
          ZIP up the folder for submission
        </p>
      </li>
    </ol>
  </Toggle>
  <Toggle
    title="IPA"
  >
    <p>
      Signing IPA files is currently Alpha. Please use xcarchive.
    </p>
  </Toggle>
  <Toggle
    title="APK"
  >
    <p>
      To package up an android package to submit for signing you need to:
    </p>
    <ol>
      <li>
        <p>
          Select RELEASE mode in the Build Variant, and build APK
        </p>
      </li>
      <li>
        <p>
          Drag and drop the apk
        </p>
      </li>
    </ol>
  </Toggle>
</div>
`;
